### PR TITLE
[1/N] Chunk encoding optimization: Support encoding format with gnark

### PR DESCRIPTION
**Pull request by jianoaix**
## Why are these changes needed?
There are inefficiency in chunk encoding, which causes more than data to transmit over network and store to db than necessary.

This PR adds the encoding with gnark. The followup PR will integrate it (compatibly).

## Checks

- [x] I've made sure the lint is passing in this PR.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: encoding/serialization.go [hunk 1]
```diff
@@ -22,6 +23,41 @@ func (c *Frame) Deserialize(data []byte) (*Frame, error) {
 	return c, err
 }
 
+func (c *Frame) SerializeGnark() ([]byte, error) {
+	coded := make([]byte, 0, 32*(1+len(c.Coeffs)))
```
Comment (bxue-l2) [inline encoding/serialization.go:27]: can replace 32 with constants.go inside encoding/*

Comment (bxue-l2) [reply encoding/serialization.go:27]: for point size you can use SizeOfG1AffineUncompressed, see https://pkg.go.dev/github.com/consensys/gnark-crypto/ecc/bn254#G1Affine

**Review by bxue-l2 — APPROVED** [review]
lgtm

File: encoding/serialization.go (hunk 1 above)
Comment (jianoaix) [reply encoding/serialization.go]: Done

File: encoding/serialization.go [hunk 2]
```diff
@@ -22,6 +23,41 @@ func (c *Frame) Deserialize(data []byte) (*Frame, error) {
 	return c, err
 }
 
+func (c *Frame) SerializeGnark() ([]byte, error) {
+	coded := make([]byte, 0, bn254.SizeOfG1AffineCompressed+BYTES_PER_SYMBOL*len(c.Coeffs))
```
Comment (bxue-l2) [inline encoding/serialization.go:27]: Do you want to consider to use SizeOfG1AffineUncompressed?  Marshal uses uncompressed format. https://github.com/consensys/gnark-crypto/blob/v0.6.1/ecc/bn254/marshal.go#L607

In current EigenDA, we use uncompressed format with 64 bytes

Comment (jianoaix) [reply encoding/serialization.go:27]: We could settle on a 32 bytes point for this new encoding format

**Pull request merged**